File: leetcode/202.happy-number/ts/happynumber.test.ts
import { expect, test } from "vitest"
import isHappy from "./happynumber.js"

interface testcase {
    n: number
    want: boolean
}

test("happynumber", () => {
    const tcs: testcase[] = [
        { n: 19, want: true },
        { n: 2, want: false }
    ]

    for (const tc of tcs) {
        expect(tc.want === isHappy(tc.n))
    }
})

File: leetcode/202.happy-number/cpp/happynumber_test.cpp
#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
#include "doctest.h"

#include "happynumber.cpp"

struct testCase
{
    int n;
    bool want;
};

TEST_CASE("HappyNumber")
{
    std::vector<testCase> tcs{
        {19, true},
        {2, false},
    };

    Solution s;

    for (auto tc : tcs)
    {
        CHECK(s.isHappy(tc.n) == tc.want);
    }
}
